document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/data/transaction/TransactionManager.java

```diff
@@ -0,0 +1,91 @@
+package uk.gov.companieshouse.document.generator.accounts.data.transaction;
+
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+import org.springframework.web.client.RestTemplate;
+import uk.gov.companieshouse.api.model.transaction.Transaction;
+import uk.gov.companieshouse.environment.impl.EnvironmentReaderImpl;
+
+/**
+ * TransactionManager is the current temporary internal project solution for communicating with
+ * microservices internally. This will be replaced by the private-sdk and its use shall be replaced
+ * in the service layer also.
+ */
+public class TransactionManager {
+
+    /** represents the Authorization header name in the request */
+    private static final String AUTHORIZATION_HEADER = "Authorization";
+
+    /** represents the Spring rest template that is created for cross microservice contact */
+    private static final RestTemplate restTemplate = createRestTemplate();
+
+    private TransactionManager() {
+        // private constructor inserted to hide implicit public one
+    }
+
+    /**
+     * Get transaction if exists
+     *
+     * @param id - transaction id
+     * @return transaction object along with the status or not found status.
+     */
+    public static ResponseEntity<Transaction> getTransaction(String id) {
+        HttpHeaders requestHeaders = new HttpHeaders();
+        requestHeaders.set(AUTHORIZATION_HEADER, getApiKey());
+
+        HttpEntity requestEntity = new HttpEntity(requestHeaders);
+        String url = getBaseUrl(id);
+
+        return getTransaction(url, requestEntity);
+    }
+
+    /**
+     * Creates the rest template when class is initialised
+     *
+     * @return Returns a statically created rest template
+     */
+    private static RestTemplate createRestTemplate() {
+        HttpComponentsClientHttpRequestFactory requestFactory = new HttpComponentsClientHttpRequestFactory();
+        return new RestTemplate(requestFactory);
+    }
+
+    /**
+     * GET the transaction resource
+     *
+     * @param url - url to send the get request
+     * @param requestEntity - the request entity object
+     */
+    private static ResponseEntity<Transaction> getTransaction(String url, HttpEntity requestEntity) {
+        return restTemplate.exchange(getRootUri() + url, HttpMethod.GET, requestEntity, Transaction.class);
+    }
+
+    /**
+     * Get the API_URL environment variable
+     *
+     * @return Get the API_URL if set, otherwise throw an exception
+     */
+    private static String getRootUri() {
+        return new EnvironmentReaderImpl().getMandatoryString("API_URL");
```

Comment:
I'm not sure this is safe. `getMandatoryString` throws a `RuntimeException` in the form of `EnvironmentVariableException` (see https://github.com/companieshouse/environment-reader-library/blob/develop/src/main/java/uk/gov/companieshouse/environment/impl/EnvironmentReaderImpl.java#L28 and https://github.com/companieshouse/environment-reader-library/blob/develop/src/main/java/uk/gov/companieshouse/environment/exception/EnvironmentVariableException.java). 

If we need a value from an environment variable, we should be getting it on startup, and failing to start the application if we can't get it. I understand the structure of this application is unique in the sense that you won't necessarily know what environment variables you'll need on startup, but that's something we probably need to look at as a problem.

That said, I feel like `API_URL` and `CHS_API_KEY` are pretty fundamental to the purpose of this API, so we should be checking these on startup, and shutting down the service if we can't get them.

Reply:
How would that work when the private sdk gets introduced? It will handle all of this as this temporarily is just an alternative until the private sdk is done. So how can we stop our service from starting by checking a variable that this module doesn't need to know about?

Reply:
As discussed, this will be completed by issue  [#14](https://github.com/companieshouse/document-generator/issues/14)